ethdb/privateapi/ethbackend.go

```diff
@@ -306,20 +306,22 @@ func (s *EthBackendServer) EngineGetBlobsBundleV1(ctx context.Context, in *remot
 
 // EngineNewPayload validates and possibly executes payload
 func (s *EthBackendServer) EngineNewPayload(ctx context.Context, req *types2.ExecutionPayload) (*remote.EnginePayloadStatus, error) {
+	fmt.Println("BC - override EngineNewPayload - Difficulty")
 	header := types.Header{
-		ParentHash:  gointerfaces.ConvertH256ToHash(req.ParentHash),
-		Coinbase:    gointerfaces.ConvertH160toAddress(req.Coinbase),
-		Root:        gointerfaces.ConvertH256ToHash(req.StateRoot),
-		Bloom:       gointerfaces.ConvertH2048ToBloom(req.LogsBloom),
-		BaseFee:     gointerfaces.ConvertH256ToUint256Int(req.BaseFeePerGas).ToBig(),
-		Extra:       req.ExtraData,
-		Number:      big.NewInt(int64(req.BlockNumber)),
-		GasUsed:     req.GasUsed,
-		GasLimit:    req.GasLimit,
-		Time:        req.Timestamp,
-		MixDigest:   gointerfaces.ConvertH256ToHash(req.PrevRandao),
-		UncleHash:   types.EmptyUncleHash,
-		Difficulty:  serenity.SerenityDifficulty,
+		ParentHash: gointerfaces.ConvertH256ToHash(req.ParentHash),
+		Coinbase:   gointerfaces.ConvertH160toAddress(req.Coinbase),
+		Root:       gointerfaces.ConvertH256ToHash(req.StateRoot),
+		Bloom:      gointerfaces.ConvertH2048ToBloom(req.LogsBloom),
+		BaseFee:    gointerfaces.ConvertH256ToUint256Int(req.BaseFeePerGas).ToBig(),
+		Extra:      req.ExtraData,
+		Number:     big.NewInt(int64(req.BlockNumber)),
+		GasUsed:    req.GasUsed,
+		GasLimit:   req.GasLimit,
+		Time:       req.Timestamp,
+		MixDigest:  gointerfaces.ConvertH256ToHash(req.PrevRandao),
+		UncleHash:  types.EmptyUncleHash,
+		// Difficulty:  serenity.SerenityDifficulty,
+		Difficulty:  big.NewInt(2),
```

Comment:
This isn't guarded by any chain type checks?

Reply:
Fixed!